base_layer/core/src/base_node/states/listening.rs

```diff
@@ -174,3 +179,96 @@ fn determine_sync_mode(
         },
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::blocks::BlockHash;
+    use std::convert::TryFrom;
+    use tari_comms::peer_manager::NODE_ID_ARRAY_SIZE;
+
+    #[test]
+    fn sync_peer_selection() {
+        let local_tip_height: u64 = 4000;
+        let network_tip_height = 5000;
+        let block_hash1: BlockHash = vec![0, 1, 2, 3];
+        let block_hash2: BlockHash = vec![4, 5, 6, 7];
+        let accumulated_difficulty1 = Difficulty::from(200000);
+        let accumulated_difficulty2 = Difficulty::from(100000);
+
+        let mut peer_metadata_list = Vec::<PeerChainMetadata>::new();
+        let best_network_metadata = best_metadata(peer_metadata_list.as_slice());
+        assert_eq!(best_network_metadata, ChainMetadata::default());
+        let sync_peers = select_sync_peers(local_tip_height, &best_network_metadata, &peer_metadata_list);
+        assert_eq!(sync_peers.len(), 0);
+
+        let node_id1 = NodeId::try_from(&[1; NODE_ID_ARRAY_SIZE][..]).unwrap();
```

Comment:
Why not just use a random NodeId?

Reply:
Change applied, made use a random NodeId.